### PR TITLE
DOC: add examples to groupby.first

### PR DESCRIPTION
**Pull request by Moisan**
- [ ] closes #xxxx (Replace xxxx with the Github issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Add a few more examples to `groupby.first`. I also removed the mention of `numeric_only = None` since the code doesn't check for None any case, nor does the type allow for it.

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -2323,12 +2322,24 @@ def first(self, numeric_only: bool = False, min_count: int = -1):
 
         Examples
         --------
-        >>> df = pd.DataFrame(dict(A=[1, 1, 3], B=[None, 5, 6], C=[1, 2, 3]))
+        >>> df = pd.DataFrame(dict(A=[1, 1, 3], B=[2, 2, None],
+        ...                        C=['3/11/2000', '3/12/2000', '3/13/2000']))
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:2326]: Could you add a `D` column instead of changing the values in the `C` columns?

**Comment by jreback** [comment]
doesn't this fix the same as #46792 ?

**Comment by Moisan** [comment]
https://github.com/pandas-dev/pandas/pull/46792 covers unused arguments in `Resampler.first` however `GroupBy.first` doc doesn't include unused arguments so I don't think that PR is related.

File: pandas/core/groupby/groupby.py [hunk 2]
```diff
@@ -2323,8 +2322,20 @@ def first(self, numeric_only: bool = False, min_count: int = -1):
 
         Examples
         --------
-        >>> df = pd.DataFrame(dict(A=[1, 1, 3], B=[None, 5, 6], C=[1, 2, 3]))
+        >>> df = pd.DataFrame(dict(A=[1, 1, 3], B=[None, 5, 6], C=[1,2,3],
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:2325]: Nit: looks like he spacing for the list assigned to `C` isn't the same as the original

**Comment by mroeschke** [comment]
@Moisan could you merge in the main branch one more time? After that LGTM

**Review by mroeschke — APPROVED** [review]

**Comment by jreback** [comment]
thanks @Moisan

**Pull request merged**